hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStoreFile.java

```diff
@@ -362,6 +369,15 @@ private void open() throws IOException {
       fileInfo.getHFileInfo().initMetaAndIndex(reader.getHFileReader());
     }
     this.initialReader = fileInfo.postStoreFileReaderOpen(context, cacheConf, reader);
+
+    if (InputStreamBlockDistribution.isEnabled(fileInfo.getConf())) {
+      boolean useHBaseChecksum = context.getInputStreamWrapper().shouldUseHBaseChecksum();
+      FSDataInputStream stream = context.getInputStreamWrapper().getStream(useHBaseChecksum);
+      if (stream instanceof HdfsDataInputStream) {
```

Comment:
We've started rolling this out to prod. During the rollout we realized that this check does not cover FileLinks, which end up as a plain FSDataInputStream wrapping a FileLinkInputStream which further wraps the real underlying HdfsDataInputStream. That's a lot of wrappers, and I'll need to unwrap them to appropriately derive HDFSBlockDistribution for FileLinks.

Reply:
This is now resolved.